routes/api/search.js

```diff
@@ -1,36 +1,95 @@
 const express = require("express");
+const Search = require("../../models/Search");
 const router = express.Router();
+const databaseControllers = require("../../services/search-repository");
 
 // @route  POST api/search
-// @desc   Add Looking For request
+// @desc   Add search request
 // @access Private
 
-router.post("/", (req, res) => {
-    res.send("Add an LF request");
+router.post("/", async ({ body }, res) => {
+    try {
+        const search = await databaseControllers.addSearch(body);
+        res.status(201).send(search);
+        return;
+    } catch (error) {
+        res.status(500).send(
+            `An error occured while trying to create your search, ${error}`
+        );
+        return;
+    }
 });
 
 // @route  DELETE api/search/:id
-// @desc   Delete your LF Request
+// @desc   Delete search request
 // @access Private
 
-router.delete("/:id", (req, res) => {
-    res.send("Delete your own LF request");
+router.delete("/:searchID", async ({ params }, res) => {
+    try {
+        const search = await databaseControllers.deleteSearch(params);
+        res.status(200).send(search);
+        return;
+    } catch (error) {
+        if (error.message === "NOT_FOUND") {
+            res.status(404).send(`Search ID does not exist`);
+            return;
+        } else {
+            res.status(500).send(
+                `An error occured while trying to delete your search, ${error}`
+            );
+            return;
+        }
+    }
 });
 
 // @route  POST api/search/:id/join/:playerID
-// @desc   Manually add someone to your request
+// @desc   Join search request
 // @access Private
 
-router.post("/:id/player/:playerID", (req, res) => {
-    res.send("Add a player to an LF Request");
+router.post("/:searchID/player/:playerID", async ({ params }, res) => {
+    try {
+        const search = await databaseControllers.joinSearch(params);
+        res.status(201).send(search);
+        return;
+    } catch (error) {
+        if (error.message === "NOT_FOUND") {
+            res.status(404).send(`Search ID does not exist`);

```

Comment:
Please don't use template literals if you're not going to interpolate values. Use regular strings instead.